torch/testing/_internal/common_utils.py

```diff
@@ -842,224 +868,245 @@ def safeCoalesce(self, t):
+                                                 self.dtype_precisions.get(b.dtype, (0, 0))[1])
+        # NOTE: legacy mechanism for setting the atol
+        #  used by @precisionOverride, for example.
+        atol = max(atol, self.precision)
+
+        # Converts to comparison dtype
+        dtype = get_comparison_dtype(a, b)
+        a = a.to(dtype)
+        b = b.to(dtype)
+
+        return _compare_tensors_internal(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
+
+    # Checks if two scalars are equal(-ish), returning (True, None)
+    #   when they are and (False, debug_msg) when they are not.
+    # NOTE: this function just acquires rtol and atol
+    #   before calling _compare_scalars_internal.
+    def _compareScalars(self, a, b, *, rtol=None, atol=None, equal_nan=True):
+        # TODO: for legacy reasons when only atol is set then rtol is
+        #   zeroed. In the future they should be acquired independently,
+        #   but this will require updating the precision values in all tests
+        #   that explicitly set atol (since they set their precision for a
+        #   test framework that didn't use rtol.)
+        rtol = 0 if ((atol is not None or self.precision != 0) and rtol is None) else rtol
+
+        # NOTE: Complex number comparisons compare the real and imaginary parts
+        #   separately, just like complex tensor comparisons.
+        if isinstance(a, complex) or isinstance(b, complex):
+            rtol = self.dtype_precisions[torch.complex64][0] if rtol is None else rtol
+            atol = self.dtype_precisions[torch.complex64][1] if atol is None else atol
+        elif isinstance(a, float) or isinstance(b, float):
+            rtol = self.dtype_precisions[torch.float32][0] if rtol is None else rtol
+            atol = self.dtype_precisions[torch.float32][1] if atol is None else atol
+        else:
+            rtol = 0 if rtol is None else rtol
+            atol = 0 if atol is None else atol
 
-        a_tol = self.get_default_tolerance(a)
-        b_tol = self.get_default_tolerance(b)
-        return (max(a_tol[0], b_tol[0]), max(a_tol[1], b_tol[1]))
+        # NOTE: legacy mechanism for setting the atol
+        #   (used by @precisionOverride, for example)
+        atol = max(atol, self.precision)
+        return _compare_scalars_internal(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
 
     def assertEqualIgnoreType(self, *args, **kwargs):
         # If you are seeing this function used, that means test is written wrongly
         # and deserves detailed investigation
         return self.assertEqual(*args, exact_dtype=False, **kwargs)
 
-    def assertEqual(self, x, y, message='', *, atol=None, rtol=None, allow_inf=False, exact_dtype=True):
+    # Compares x and y
+    # TODO: make message kwarg-only
+    # TODO: default exact_device to True
+    def assertEqual(self, x, y, message=None, *, atol=None, rtol=None, equal_nan=True,
+                    exact_dtype=True, exact_device=False):
```

Comment:
@zou3519 or @gchanan any second opinion on the removal of allow_inf? I don't necessarily have strong objections but I'm not sure what the history is on the flag.

I think the case this would affect is if someone tried to write a test with values outside the range of a floating type (e.g. a half greater than 65504) that converted to inf, the test would have rejected the comparison (by default). Presumably no such tests currently exist (or they would have failed), but maybe the point was to avoid adding such low-signal tests.

The flag was recently inadvertently broken on master and I had been planning to fix it, this PR removes it instead.

Reply:
This flag was always broken (at first it was comparing -inf and inf as equal if you allowed inf, and then, as you point out, it was inadvertently ignored). 
The behavior you describe (failing low precision tests that result in overflow) can be achieved by having wider-precision reference that would not overflow. The current state where no such tests exist proves that it is a niche scenario. IMO it does not make sense to support such niche scenarios in a general-purpose function, so explicitly removing the flag and always allowing tensors containing infs to compare equal is fine.